lib/pre2017.js

```diff
@@ -0,0 +1,109 @@
+const fetch = require('node-fetch')
+const cheerio = require('cheerio')
+const https = require('https')
+
+const PRE_URL = [
+  'https://codein.withgoogle.com/archive/2016',
+  'https://codein.withgoogle.com/archive/2015',
+  'https://www.google-melange.com/archive/gci/2014', // Ignore InvalidLinkBear
+  'https://www.google-melange.com/archive/gci/2013', // Ignore InvalidLinkBear
+  'https://www.google-melange.com/archive/gci/2012', // Ignore InvalidLinkBear
+  'https://www.google-melange.com/archive/gci/2011', // Ignore InvalidLinkBear
+  'https://www.google-melange.com/archive/gci/2010', // Ignore InvalidLinkBear
+]
+
+async function fetchText(url) {
+  const agent = new https.Agent({
+    rejectUnauthorized: false,
+  })
+  const res = await fetch(url, {
+    agent,
+  })
+
+  return await res.text()
+}
+
+async function fetchPre2017(url, date) {
+  const year = url.split('/').pop()
+  const responseHTML = await fetchText(url)
+
+  return await scrape(responseHTML, year, date)
+}
+
+async function scrape(responseHTML, year, date) {
```

Comment:
1. export this function
2. call it from the test
3. provide 2 "fake" data; html & json result from the scrapping process
3. assert the resulting HTML matches json result

Reply:
Or you could mock the `fetch` and test the function in `module.exports`

That will test the whole file, but it is too complex.

Reply:
so I need to compare scraped data from url and scraped data from 'mocked' html?

Reply:
This function takes html string, right? Use that downloaded html to call the function.

Roughly the test will be like this
```js
const res = await scrape(html, year)
expect(res).toEqual(expectedResult)
```

where the `expectedResult` contains the expected result of the function call with that corresponding html.